### ISSUE TITLE
Storing b32 usernames on device file system

### ISSUE THREAD

**Issue by devgeeks** [SpiderOak/SpiderOakMobileClient#128]
_tl;dr: Is it bad to store the b32 username as part of a file path on a stored file, or is that leaking information that could be used for badness?_
## 

Currently, when saving a favorite, the path on the device is derived from the URL of the item.

So if the item being saved has a URL of:

`https://web-dc2.spideroak.com/storage/MGSXXZ6FMVVXY/DeviceName/Users/tommy/Pictures/best-pic-evar.png`

...then the file is saved on the device as:

`Download/SpiderOak/.favorites/DeviceName/Users/tommy/Pictures/best-pic-evar.png`

**However**

It's certainly possible that two users of the same device might have a device name that is the same (i.e.: "My Mac"), or that a device name and an initial folder name of a ShareRoom might be the same. This would lead to a possible collision.

A user meeting those criteria could end up favoriting a file that would then overwrite a previous favorite.

So here's the QUESTION part:

Is it bad to store the b32 username as part of the file path to alleviate this, or is that leaking information that could be used for badness?

//cc @alanfairless
